src/test/ui/error-codes/E0730.rs

```diff
@@ -3,7 +3,7 @@
 
 fn is_123<const N: usize>(x: [u32; N]) -> bool {
     match x {
-        [1, 2, 3] => true, //~ ERROR cannot pattern-match on an array without a fixed length
```

Comment:
This is the only place this error occurs in the test suite, so let's add a new UI test in `src/test/ui/array-slice-vec/` for the error with a subslice pattern.

Reply:
updated this test to emit E0730 again,
copied the old version to array-slice-vec